source/docs/software/examples-tutorials/drivesim-tutorial/simulation-instance.rst

```diff
@@ -0,0 +1,69 @@
+Step 1: Creating Simulated Instances of Hardware
+================================================
+The WPILib simulation framework contains several ``XXXSim`` classes, where ``XXX`` represents physical hardware such as encoders or gyroscopes. These simulation classes can be used to set positions and velocities (for encoders) and angles (for gyroscopes) from a model of your drivetrain.
+
+Simulating Encoders
+-------------------
+The ``EncoderSim`` class allows users to set encoder positions and velocities on a given ``Encoder`` object. When running on real hardware, the ``Encoder`` class interacts with real sensors to count revolutions (and convert them to distance units automatically if configured to do so); however, in simulation there are no such measurements to make. The ``EncoderSim`` class can accept these simulated readings from a model of your drivetrain.
+
+.. tabs::
+   .. code-tab:: java
+
+      // These represent our regular encoder objects, which we would
+      // create to use on a real robot.
+      private Encoder m_leftEncoder = new Encoder(0, 1);
+      private Encoder m_rightEncoder = new Encoder(2, 3);
+
+      // These are our EncoderSim objects, which we will only use in
+      // simulation. However, you do not need to comment out these
+      // declarations when you are deploying code to the roboRIO.
+      private EncoderSim m_leftEncoderSim = new EncoderSim(m_leftEncoder);
+      private EncoderSim m_rightEncoderSim = new EncoderSim(m_rightEncoder);
+
+   .. code-tab:: c++
+
+      #include <frc/Encoder.h>
+      #include <frc/simulation/EncoderSim.h>
+
+      ...
+
+      // These represent our regular encoder objects, which we would
+      // create to use on a real robot.
+      frc::Encoder m_leftEncoder{0, 1};
+      frc::Encoder m_rightEncoder{2, 3};
+
+      // These are our EncoderSim objects, which we will only use in
+      // simulation. However, you do not need to comment out these
+      // declarations when you are deploying code to the roboRIO.
+      frc::sim::EncoderSim m_leftEncoderSim{m_leftEncoder};
+      frc::sim::EncoderSim m_rightEncoderSim{m_rightEncoder};
+
+Simulating Gyroscopes
+---------------------
+Similar to the ``EncoderSim`` class, simulated gyroscope classes also exist for commonly used WPILib gyros -- ``AnalogGyroSim`` and ``ADXRS450_GyroSim``. These are also constructed in the same manner.
```

Comment:
I think Gyros such as the NavX and Pigeon are extremely popular. Should we add a note that we can't simulate these?